editor/scene_tree_editor.cpp

```diff
@@ -688,8 +688,6 @@ void SceneTreeEditor::set_selected(Node *p_node, bool p_emit_selected) {
 		tree->ensure_cursor_is_visible();
 
 	} else {
-		if (!p_node)
```

Comment:
The branch is not redundant, `item` can be `NULL` while `p_node` is not, if the `find()` call returns `NULL`.

That being said, the `selected = p_node;` below is equivalent to `selected = NULL;` here when `p_node` is `NULL`, so this change might still be OK *if* `_update_tree()` doesn't rely on `selected` already being `NULL` when it's called.

Reply:
You're right. I guess the assumption is that _update_tree() does not rely on `selected`, will investigate further.